AEPTarget/Sources/TargetState.swift

```diff
@@ -123,7 +123,7 @@ class TargetState {
         }
         if
             let newClientCode = configuration[TargetConstants.Configuration.SharedState.Keys.TARGET_CLIENT_CODE] as? String,
-            newClientCode != clientCode
+            storedConfigurationSharedState != nil, newClientCode != clientCode
```

Comment:
storedConfigurationSharedState will be nil only after close & launch. Subsequently it will be populated (see L131) and this check https://github.com/adobe/aepsdk-target-ios/blob/dev-v3.2.0/AEPTarget/Sources/Target.swift#L68 in readyForEvent will ensure it's present for subsequent requests.